network-config-analyzer/TcpLikeProperties.py

```diff
@@ -120,155 +124,66 @@ def get_properties_obj(self):
 
     def __or__(self, other):
-        res = TcpLikeProperties()
-        res.cubes_set = self.cubes_set | other.cubes_set
-
-        res.named_ports = dict({})
-        for port_name in self.named_ports:
-            res.named_ports[port_name] = self.named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res.named_ports:
-                res.named_ports[port_name] |= other.named_ports[port_name]
-            else:
-                res.named_ports[port_name] = other.named_ports[port_name]
-
-        res.excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res.excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res.excluded_named_ports:
-                res.excluded_named_ports[port_name] -= other.named_ports[port_name]
-
+        res = self.copy()
+        res |= other
         return res
 
     def __sub__(self, other):
-        res = TcpLikeProperties()
-        res.cubes_set = self.cubes_set - other.cubes_set
-
-        res.named_ports = dict({})
-        for port_name in self.named_ports:
-            res.named_ports[port_name] = self.named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res.named_ports:
-                res.named_ports[port_name] -= other.named_ports[port_name]
-
-        res.excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res.excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res.excluded_named_ports:
-                res.excluded_named_ports[port_name] |= other.named_ports[port_name]
-            else:
-                res.excluded_named_ports[port_name] = other.named_ports[port_name]
-
+        res = self.copy()
+        res -= other
         return res
 
     def __iand__(self, other):
-        self.cubes_set &= other.cubes_set
-
-        res_named_ports = dict({})
-        for port_name in self.named_ports:
-            if port_name in other.named_ports:
-                src_interval_res = self.named_ports[port_name] & other.named_ports[port_name]
-                res_named_ports[port_name] = src_interval_res
-        self.named_ports = res_named_ports
-
-        res_excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res_excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.excluded_named_ports:
-            if port_name in res_excluded_named_ports:
-                res_excluded_named_ports[port_name] |= other.excluded_named_ports[port_name]
-            else:
-                res_excluded_named_ports[port_name] = other.excluded_named_ports[port_name]
-        self.excluded_named_ports = res_excluded_named_ports
-
+        assert not self.excluded_named_ports
```

Comment:
can replace the three assertions here with the following two assertions:
```
assert not self.has_named_ports()
assert not isinstance(other, TcpLikeProperties) or not other.has_named_ports()
```

Reply:
Done